product-service/data/repositories/products-repository-impl.ts

```diff
@@ -0,0 +1,49 @@
+import {Client, QueryResult} from "pg";
+import {Product, ProductsRepository} from "../../interfaces";
+
+export class ProductsRepositoryImpl implements ProductsRepository {
+  constructor(private db: Client) {}
+
+  async findAll(): Promise<Product[]> {
+    const result: QueryResult<Product> = await this.db.query(`
+      SELECT p.*, s.count FROM products AS p
```

Comment:
Just a note - it is not recommended to use asterisks in SQL select queries (except for ad-hoc queries in DB console, of course) 
see - https://stackoverflow.com/questions/15250812/using-asterisk-in-select-queries